model/task/task.go

```diff
@@ -1767,15 +1767,26 @@ func (t *Task) MarkUnscheduled() error {
 
 }
 
-func (t *Task) MarkUnattainableDependency(dependency *Task, unattainable bool) error {
+// MarkUnattainableDependency updates the unattainable field for the dependency in the task's dependency list,
+// and logs if the task is newly blocked.
+func (t *Task) MarkUnattainableDependency(dependencyId string, unattainable bool) error {
+	wasBlocked := t.Blocked()
 	// check all dependencies in case of erroneous duplicate
 	for i := range t.DependsOn {
-		if t.DependsOn[i].TaskId == dependency.Id {
+		if t.DependsOn[i].TaskId == dependencyId {
 			t.DependsOn[i].Unattainable = unattainable
 		}
 	}
 
-	return UpdateAllMatchingDependenciesForTask(t.Id, dependency.Id, unattainable)
+	if err := updateAllMatchingDependenciesForTask(t.Id, dependencyId, unattainable); err != nil {
+		return err
+	}
+
+	// only want to log the task as blocked if it wasn't already blocked
+	if !wasBlocked && unattainable {
+		event.LogTaskBlocked(t.Id, t.Execution)
```

Comment:
Does this mean that the task will never run? Can it at some point become attainable again once the dependency gets unblocked?

Reply:
This just means that it's currently blocked. It can become attainable again if the dependency is unblocked, however we aren't displaying this log message anywhere, we're just using it to notify the user that the task _is_ blocked so they can intervene if they want (I don't think adding notifications for a task becoming unblocked is really the ask of this ticket).

Reply:
Ah that makes sense. I was thinking that we may not want to notify if something is blocked temporarily (and will be unblocked a minute later), but if the user may be able to intervene, maybe they actually want that.